effects/sleep.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import time
+
+import click
+
+from rpi_ws281x import PixelStrip, Color
+
+# LED strip default configuration:
+#LED_COUNT = 50       # Number of LED pixels.
+LED_PIN = 18          # GPIO pin connected to the pixels (18 uses PWM!).
+LED_FREQ_HZ = 800000  # LED signal frequency in hertz (usually 800khz)
+LED_DMA = 10          # DMA channel to use for generating signal (try 10)
+LED_BRIGHTNESS = 255  # Set to 0 for darkest and 255 for brightest
+LED_INVERT = False    # True to invert the signal (when using NPN transistor level shift)
+LED_CHANNEL = 0       # set to '1' for GPIOs 13, 19, 41, 45 or 53
+
+@click.command()
+@click.option('-l', '--leds', type=int, default=50, show_default=True, help='Amount of LEDs in the string')
+@click.option('-s', '--speed', type=float, default=1.0, show_default=True, help='Speed of the effect')
+@click.option('-c', '--color', type=str, default='FF0000', show_default=True, help='RGB Value for the effect. You need to provide this value as a hexadecimal.')
+def main(leds: int, speed: float, color: str):
+    '''
+    This is a sleep effect which provide a light for the night
+    '''
+    if not color.startswith('0x'):
+        color = '0x' + color
+        
+    print(f'Sleep effect started with following parameter: leds={leds}, speed={speed}, color={color}')
+    color_dict = parseColorString(color)
+    
+    strip = PixelStrip(leds, LED_PIN, LED_FREQ_HZ, LED_DMA, LED_INVERT, LED_BRIGHTNESS, LED_CHANNEL)
+    strip.begin() # Intialize the library (must be called once before other functions).
+    
+    runEffect(strip, speed, color_dict['red'], color_dict['green'], color_dict['blue']);
+    pass
+
+def parseColorString(color: str):
+    '''
+    Parse the color given as string to single RGB colors as int values and return the dictionary with
+    parsed values
+    '''
+    color_int_val = int(color, base=16)
+    
+    return {
+        'red': (color_int_val >> 16) & 0xFF,
+        'green': (color_int_val >> 8) & 0xFF,
+        'blue': color_int_val & 0xFF
+    }
+
+def runEffect(strip, speed, red_max_val, green_max_val, blue_max_val):
```

Comment:
* This effect faces out LED by LED, not all LEDs at once. Is this the intended way?
* I really don't understand what you intend to do. For a better review, please describe the desired effect. I think we should discuss this in person.

Reply:
LED 0 Is fading slow in ... LED 0 Is fading slow out while LED 1 fading slow in ... LED 1 fading slow out while LED 2 fading slow in ... 

So you have defined color moving slow along the whole strip